src/commands/apphosting-secrets-describe.ts

```diff
@@ -0,0 +1,29 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { needProjectId } from "../projectUtils";
+import { logger } from "../logger";
+import { requireAuth } from "../requireAuth";
+import { listSecretVersions } from "../gcp/secretManager";
+import * as secretManager from "../gcp/secretManager";
+import { requirePermissions } from "../requirePermissions";
+
+const Table = require("cli-table");
+
+export const command = new Command("apphosting:secrets:describe <secretName>")
+  .description("Get metadata for secret and its versions.")
+  .before(requireAuth)
+  .before(secretManager.ensureApi)
+  .before(requirePermissions, ["secretmanager.secrets.get"])
+  .action(async (secretName: string, options: Options) => {
+    const projectId = needProjectId(options);
```

Comment:
It looks like `apphosting:secrets:describe`, `functions:secrets:describe`, and `functions:secrets:get` are all identical commands. Big picture, the API could maybe be simplified to a single `secrets:describe` command. 

Small picture, we should define this action function once and use it in all 3 commands.

Reply:
Agreed re:long term, this is a priority in the future as we continue to merge the functions and apphosting secrets functionality.

`apphosting:secrets:describe` has a different table so going to keep it as is but I'll extract the `functions:secrets` stuff out